sound/soc/intel/boards/sof_sdw_hdmi.c

```diff
@@ -52,6 +52,12 @@ int sof_sdw_hdmi_card_late_probe(struct snd_soc_card *card)
 	int err, i = 0;
 	char jack_name[NAME_SIZE];
 
+	if (list_empty(&ctx->hdmi_pcm_list))
+		return -EINVAL;
```

Comment:
Under which conditions can this happen? Is this really related to this patch? Does this patch really add a possibility for this function to be called with this list empty? And even if this does now become possible: shouldn't this be `ENODEV`?

Reply:
@lyakh Fair point that this should perhaps be in a separate patch. This same check is in other machine drivers (and EINVAL is used in those) and protects against the case where topology does not have PCMs for HDMI. In this case, driver should be fail gracefully and not crash. Let me split this out into a separate patch.

Reply:
@lyakh Now moved to a separate branch, please check.